CONTRIBUTING.md

```diff
@@ -58,10 +61,34 @@ of both the contributors and the project.
 This project follows the
 [Google Python style guide](https://google.github.io/styleguide/pyguide.html).
 
-Additionally, any contributions must pass the linter `scripts/lint.sh` when executed from a pipenv shell (i.e. after running `pipenv shell`). To run the linter on changed files only, commit your changes and run `scripts/lint.sh --diff`.
+Additionally, any contributions must pass the linter `scripts/lint.sh` when executed from a
+pipenv shell (i.e. after running `pipenv shell`). To run the linter on changed files only,
+commit your changes and run `scripts/lint.sh --diff`.
 
 Finally, all python files in the repository must display the copyright of the project,
 to protect the terms of the license. Please make sure that your files start with a line like:
 ```
 # Copyright 20xx-present Kensho Technologies, LLC.
 ```
+
+## Python 2 vs Python 3
+
+In order to ensure that tests run with a fixed set of packages in both Python 2 and Python 3,
+we always run the tests in a virtualenv managed by pipenv. However, since some of our dependencies
+have different requirements for Python 2 and Python 3, we have to keep two pipenv lockfiles -- one
+per Python version.
+
+We have chosen to make the Python 3 lockfile the default (hence named `Pipfile.lock`),
+since Python 3 offers better performance and we like our tests and linters running quickly.
+The Python 2 lockfile is named `Pipfile.py2.lock`.
+
+If you need to set up a Python 2 virtualenv locally, simply run the following script:
+```
+./scripts/make_py2_venv.sh
+```
+
+If you change the Pipfile or the package requirements, please make sure to regenerate the
+lockfiles for both Python versions. The easiest way to do so is with the following script:
+```
+./scripts/make_pipenv_lockfiles.sh
+```
```

Comment:
What kind of error will developers get when they forget to do this?

Reply:
We install dependencies for tests using `pipenv install --dev --deploy --system (see the `.travis.yml` file), which promises to assert that the lockfile and Pipfile match up:

```
--system                 System pip management.  [env var: PIPENV_SYSTEM]
--deploy                 Abort if the Pipfile.lock is out-of-date, or Python
                           version is wrong.
```